Konductor responds with RFC 7807 for fatal errors

raresvla opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description

<!--- Describe your changes in detail -->
- Fixed tests to properly assert the new error response
- Opt-out is no longer considered an error (is a warning)

## Related Issue

<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

## Motivation and Context

<!--- Why is this change required? What problem does it solve? -->

Updates test asserts to comply with upcoming Konductor changes.

## Screenshots (if appropriate):

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [x] I have made any necessary test changes and all tests pass.
- [ ] I have run the Sandbox successfully.

Aaronius commented:
Hey @raresvla, just so you're aware, you should have admin rights to this repo so you don't have to work on your personal fork. In fact, we encourage you working on the main fork because it's easier for us to check out and add commits to your branch.

Aaronius commented on test/functional/specs/Privacy/IAB/C224677.js:
```diff
@@ -67,11 +65,17 @@ test("Test C224677: Call setConsent when purpose 10 is FALSE", async () => {
 
-  // Event calls going forward should be Opted-Out because AAM opts out consents with no purpose 10.
-  const errorMessage = await getErrorMessageFromSendEvent();
+  // 3. Event calls going forward should be Opted-Out because AAM opts out consents with no purpose 10.
```
Grammatically, "Opted-Out" here should be "opted out" (it's the past tense of a transitive verb).

Aaronius commented on test/functional/specs/Privacy/IAB/C224677.js:
```diff
@@ -67,11 +65,17 @@ test("Test C224677: Call setConsent when purpose 10 is FALSE", async () => {
+  const eventResponse = createResponse(rawEventResponse);
 
-  await t.expect(errorMessage).contains("EXEG-0301-403");
+  // 4. And a warning message should be returned, confirming the Opted-Out
```
Grammatically, "Opted-Out" here should be "opt-out" (it's a noun).

Aaronius commented on test/functional/specs/Privacy/IAB/C224678.js:
```diff
@@ -97,7 +96,11 @@ test.skip("Test C224678: Passing a negative Consent in the sendEvent command", a
     .notOk("Event returned an error when we expected it not to.");
 
-  // 5. Events should be blocked going forward because we are opted out.
+  // 5. But returns a warning message confirming the Opted-Out
```
"Opted-Out" should be "opt-out"

Aaronius reviewed (approved):
I believe this all makes sense to me and looks good. Thanks for making the updates @raresvla!

raresvla commented:
@Aaronius I've addressed your comments, thank you! 

@jfkhoury / @shammowla -- I see build that is failing here, although is passing locally. Is this a know issue?

jfkhoury reviewed (approved):
Thank you @raresvla !

@shammowla Can you please look into the tests?

shammowla commented:
Hey @raresvla, I was able to pinpoint issue with the tests failing on GH actions. I cloned your commits and added it to this PR: https://github.com/adobe/alloy/pull/642

The required keys and environment variables didn't pass to GH actions. Also, the unit tests karma runner was creating a separate tunnel to browserstack which caused the unit tests to fail. 

Both the unit tests and functional test passed with the mentioned changes. 
https://github.com/adobe/alloy/actions/runs/409042942

Aaronius commented:
@raresvla, thanks for your help on this PR. I'm going to close it since it's been superseded by https://github.com/adobe/alloy/pull/642 (your changes are reflected there).
